packages/sdk/src/sdk/services/Storage/Storage.ts

```diff
@@ -197,6 +127,34 @@ export class Storage implements StorageService {
     )
   }
 
+  /**
+   * Generates a preview for a track at the given second offset
+   * @param {Object} params
+   * @param {string} params.cid - The CID of the track to generate a preview for
+   * @param {number} params.secondOffset - The offset in seconds to start the preview from
+   * @returns {Promise<string>} The CID of the generated preview
+   */
+  async generatePreview({
+    cid,
+    secondOffset
+  }: {
+    cid: string
+    secondOffset: number
+  }) {
+    const contentNodeEndpoint = await this.storageNodeSelector.getSelectedNode()
+
+    if (!contentNodeEndpoint) {
+      throw new Error('No content node available')
+    }
+
+    const response = await axios({
```

Comment:
Wasn't aware the storage node used axios....

Reply:
Yeah we have to use it because `fetch` doesn't support upload progress and raw XMLHttpRequest isn't supported on node

Wanted to stay consistent inside Storage node for the generate_preview function